models/sudoswap/ethereum/sudoswap_ethereum_pools.sql

```diff
@@ -0,0 +1,275 @@
+          1
+      ),
+      eth_out AS (
+        SELECT
+          tr.FROM AS holder_address,
+          SUM(tr.value / 1e18) AS eth_spent
+        FROM
+          {{ source('ethereum','traces') }} tr
+          JOIN pairs_created pc ON pc.pair_address = tr.FROM
+        WHERE
+          tr.block_time > '2022-04-23'
+          AND tr.success = true
+          AND tr.type = 'call'
+          AND (
+            tr.call_type NOT IN ('delegatecall', 'callcode', 'staticcall')
+            OR tr.call_type IS null
+          )
+          {% if is_incremental() %}
+          -- this filter will only be applied on an incremental run. We only want to update with new traces.
+          AND tr.block_time >= date_trunc("day", now() - interval '1 week')
+          {% endif %}
+        GROUP BY
+          1
+      )
+    SELECT
+      eth_in.holder_address,
+      eth_in.eth_funded,
+      eth_out.eth_spent,
+      COALESCE(eth_funded, 0) - COALESCE(eth_spent, 0) as eth_balance
+    FROM
+      eth_in
+      LEFT JOIN eth_out ON eth_in.holder_address = eth_out.holder_address
+  ),
+  all_pairs_cleaned AS (
+    SELECT
+      pricing_type,
+      pool_type,
+      nft_contract_address,
+      creator_address,
+      COALESCE(mr.delta, pc.delta) AS delta --if delta was never updated, just keep original deploy delta
+,
+      round(COALESCE(mr.spot_price, pc.spot_price), 4) as spot_price --same logic as above
+,
+      nft_bal.tokens_held,
+      round(COALESCE(eth_bal.eth_balance, 0), 4) as eth_balance,
+      pc.pair_address AS raw_pair_address,
+      days_passed,
+      pc.spot_price AS initial_price,
+      pc.initial_nft_count,
+      pc.initial_eth
+    FROM
+      pairs_created pc
+      LEFT JOIN most_recent_spot_delta mr ON pc.pair_address = mr.pair_address
+      LEFT JOIN erc721_balances nft_bal ON nft_bal.holder_address = pc.pair_address
+      LEFT JOIN eth_balances eth_bal ON eth_bal.holder_address = pc.pair_address
+  ),
+  trading_totals as (
+    SELECT
+      CASE
+        WHEN trade_category = 'Sell' THEN buyer
+        ELSE seller
+      END AS pair_address,
+      sum(amount_original) AS eth_volume,
+      sum(amount_usd) AS usd_volume,
+      sum(number_of_items) AS nfts_traded,
+      sum(pool_fee_amount) AS owner_fee_volume_eth,
+      sum(platform_fee_amount) AS platform_fee_volume_eth,
+      sum(
+        CASE
+          WHEN trade_category = 'Sell' THEN -1 * amount_original
+          ELSE (amount_original-platform_fee_amount)
+        END
+      ) AS eth_change_trading,
+      sum(
+        CASE
+          WHEN trade_category = 'Sell' THEN number_of_items
+          ELSE -1 * number_of_items
+        END
+      ) AS nft_change_trading
+    FROM
+      (
+        SELECT
+          trade_category,
+          buyer,
+          seller,
+          amount_original,
+          amount_usd,
+          number_of_items,
+          pool_fee_amount,
+          platform_fee_amount
+        FROM
+          ({{ ref('sudoswap_ethereum_events') }})
```

Comment:
imagine this needs incremental filters as well

Reply:
ok ive added one now